kyo-prelude/shared/src/main/scala/kyo/Emit.scala

```diff
@@ -22,8 +22,10 @@ object Emit:
       */
     opaque type Ack = Int
     object Ack:
-        given CanEqual[Ack, Ack] = CanEqual.derived
-        inline given Flat[Ack]   = Flat.unsafe.bypass
+        given CanEqual[Ack, Ack]         = CanEqual.derived
+        inline given Flat[Ack]           = Flat.unsafe.bypass
+        inline given Flat[Ack.Stop.type] = Flat.unsafe.bypass
+        inline given Flat[Ack.Continue]  = Flat.unsafe.bypass
```

Comment:
Since `Flat` is invariant, we have to provide explicit evidences for sub-classes. I've also tried making `Flat` covariant but I had trouble with the compiler inferring the wrong necessary evidence in a few scenarios, especially with `Async` operations.